tests/system/feature_store/test_spark_azure.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+
+import pandas as pd
+import pytest
+
+import mlrun.feature_store as fstore
+from mlrun import store_manager
+from mlrun.datastore.datastore_profile import (
+    DatastoreProfileAzureBlob,
+    register_temporary_client_datastore_profile,
+)
+from mlrun.datastore.sources import ParquetSource
+from mlrun.datastore.targets import (
+    ParquetTarget,
+)
+from tests.system.base import TestMLRunSystem
+from tests.system.feature_store.expected_stats import expected_stats
+
+
+@TestMLRunSystem.skip_test_if_env_not_configured
+# Marked as enterprise because of v3io mount and remote spark
+@pytest.mark.enterprise
+class TestFeatureStoreAzureSparkEngine(TestMLRunSystem):
```

Comment:
Seems to me like duplicating the original test suite, with all of its utilities such as `run_local`, `spark_image_deployed`, `test_branch`, etc, is not a good idea. Better make a common base class. This common base would of course undo the duplication added in #4787 for S3.

Reply:
On this matter, I hold a different viewpoint. When I attempted to augment `test_spark_engine.py` with S3 integration, the outcome was overly complex and challenging to comprehend or maintain. Subsequently, it dawned on me that our primary focus is testing the setup, such as the need for additional installations like `azure-storage-blob` and varying datastore profiles, rather than the test itself. The apparent duplication in the test is essentially just the necessary boilerplate.

Reply:
Well, I definitely wouldn't augment the original class, but rather extract a base to deduplicate about 100 lines of code and documentation that are duplicated per suite (potentially including the test itself, which is almost identical). Otherwise, it's x3 duplication, with the potential to grow (e.g. for gcs).

And even failing that, I can't see why we would want to duplicate the docstring verbatim.

Reply:
Will try to do something about